packages/provider/src/eip191exceptions.ts

```diff
@@ -0,0 +1,123 @@
+import { ethers } from 'ethers'
+
+export function messageIsExemptFromEIP191Prefix(message: Uint8Array, origin: string | undefined): boolean {
+  return EIP_191_PREFIX_EXCEPTIONS.some(e => e.predicate(message, origin))
+}
+
+const EIP_191_PREFIX_EXCEPTIONS: Array<{
+  name: string
+  predicate: (message: Uint8Array, origin?: string) => boolean
+}> = [
+  // NOTE: Decentraland does not support 191 correctly.
+  { name: 'Decentraland Exception', predicate: (_message, origin) => origin === 'https://play.decentraland.org' },
+
+  // NOTE: 0x v3 does not support 191 correctly.
+  // See https://gov.0x.org/t/zeip-proposal-fix-v3-eip-191-non-compliance-when-validating-eip-1271-signatures/3396 for more info.
+  { name: '0x v3 Exception', predicate: isZeroExV3Order }
+]
+
+// try to interpret bytes as abi-encoded 0x v3 OrderWithHash -
+// see https://github.com/0xProject/0x-protocol-specification/blob/master/v3/v3-specification.md
+export function isZeroExV3Order(bytes: Uint8Array): boolean {
+  const abi = new ethers.utils.Interface(ZeroXV3EIP1271OrderWithHashAbi)
+  try {
+    abi.decodeFunctionData('OrderWithHash', bytes)
```

Comment:
Is this the best we can do to filter out 0x orders? maybe there is also something inside the parsed order that we can also validate

Reply:
this does decode all the way to a 0x order - could verify that the hash matches the order, though, sound good?

Reply:
Thinking harder about this I don't think it's *that* necessary, but it shouldn't hurt. If it isn't much work to add it then it sounds good.

Reply:
we can't verify the hash is correct, since we don't know what chain they're signing the typed data for, so we can't recreate the same hash